### DISCUSSION TITLE
Batch Editor Not Randomizing Tera Type (+sort all)

### DISCUSSION THREAD

**Discussion by ewormPL** [kwsch/PKHeX#3698] [Q&A]
Trying to use:

.TeraTypeOverride=$rand

in the Batch Editor. The result is "modified x files", but the Tera Types do not in fact change. I can still use sth like:

.TeraTypeOverride=Ground

and it works, but the $rand in particular doesn't seem to work in this case.

(This is an aside, but I can't see an option to sort all Pokémon in all boxes at once? Would it be possible to add the "Sort Boxes" options but as "Sort All"?

**Comment by kwsch** [comment]
1) use `.TeraTypeOverride=$1,18` or whatever the random range span you want. $rand is for hardcoded cases only.
2) refer to the shortcuts. Hold shift when clicking a box action to do it to all boxes instead of just the current box.

**Reply by ewormPL** [reply]
> use .TeraTypeOverride=$1,18 or whatever the random range span you want. $rand is for hardcoded cases only.

**Doesn't work.** In fact **the numerical values have no effect** on the TeraTypeOriginal or TeraTypeOverride at all. The only argument that works is the name of the type, like Bug or Fighting.

> refer to the shortcuts. Hold shift when clicking a box action to do it to all boxes instead of just the current box.

Thanks!

EDIT:
Also, is there a way to move Pokémon between boxes other than selecting a mon, using "set" on an empty slot and then deleting the original, one at a time? Seems very cumbersome. As far as I can tell, `.Box=` can't be used in the batch editor to move all Pokémon meeting chosen condition(s) to a particular box?
